opencog/guile/SchemePrimitive.h

```diff
@@ -30,640 +78,231 @@ class PrimitiveEnviron
-//!
-//! This template may be used to wrap a C++ object in such a way that it
-//! can be invoked from scheme code.  The file "PrimitiveExample.cc"
-//! provides a detailed example of how to do this, and how to invoke the
-//! resulting primitive.
-//!
-//! This template has a handful of pre-defined signatures. If you cannot
-//! find the signature that you need, you will need to extend this
-//! template a bit to add the needed signature. This shouldn't be hard;
-//! just work from the existing examples; please keep things in
-//! alphabetical order.
-//
-// Users and abusers of this file:
-//
-// There is a whole lot of abuse of this file, by numerous extremely
-// poorly designed scheme bindings. Some are OK and just fine, but
-// many of these users need to be fixed.  It was never the intent
-// that this would become a free-for-all, anything-goes dumping ground
-// for badly designed API's. But that is what it has become.  Time
-// to reverse the decay.
-//
-// Here is a list of some of the users:
-//
-// B_B    -- cogutils logger boolean setters/getters.
-// B_HH   -- PatternSCM::value_is_type(), etc.
-//        -- LGDictSCM::do_lg_conn_type_match(), etc.
-// H_HH   -- ?? someoene??
-// H_HHH  -- pointmem (the 3D spatial API)
-// H_HHHH -- do_backward_chaining, do_forward_chaining
-// H_HT   -- fetch-incoming-by-type
-// H_HZ   -- cog-bind-first-n
-// I_V    -- cogutils RandGen
-// P_H    -- FunctionWrapper
-// S_AS   -- CogServerSCM::start_server()
-// S_S    -- cogutils logger API, see guile/LoggerSCM.h
-// S_SS   -- DistSCM  (Gearman server)
-// S_V    -- CogServerSCM::stop_server()
-//        -- cogutils logger get_level(), etc.
-// V_B    -- SQLPersistSCM:do_set_stall, PatternMiner
-// V_I    -- cogutils do_randgen_set_seed
-// V_II   -- SQLPersistSCM:do_set_hilo
-// V_S    -- cogutils logger setters
-//        -- SQLPersistSCM::do_open()
-//        -- ZMQPersistSCM::do_open()
-// V_V    -- SQLPersistSCM::do_close(), do_load(), do_store()
-//        -- WordSenseProcessor::run_wsd()
-//
-// Users that probably should be fixed:
-// V_SA   -- PythonSCM::apply_as
-//           That's backwards, should probably be V_AS
-// K_H    -- SuRealSCM::do_non_cached_sureal_match(), etc.
-//           Should be re-written to use H_H not K_H
-//
-// This API needs to be re-thought, from scratch. Its offensive.
-// H_HTQB -- FuzzySCM::do_nlp_fuzzy_match()
-//
-// This API needs to be re-thought, from scratch. Its offensive.
-// Its complete and total crap.
-// D_HHTB -- DimEmbedModule::euclidDist()
-// Q_HTIB -- DimEmbedModule::kNearestNeighbors()
-// V_T    -- DimEmbedModule::logAtomEmbedding()
-// V_TI   -- DimEmbedModule::embedAtomSpace()
-// V_TIDI -- DimEmbedModule::addKMeansClusters
-//
-// This API needs to be re-thought, from scratch. Its offensive.
-// S_AS   -- PatternMiner
-// S_B    -- PatternMiner
-// S_I    -- PatternMiner
-// S_V    -- PatternMiner
-// V_SI   -- PatternMiner
-
```

Comment:
This list should be restored. It lists all the big users and abusers of the API: its the master "naughty and nice" list, and it took a LOT of work to asemble.  I don't want to loose this vital info.